test/apps/iodemo/io_demo.cc

```diff
@@ -1366,16 +1375,20 @@ class DemoClient : public P2pDemoCommon {
         LOG << "terminate connection " << server_info.conn << " due to "
             << reason;
 
-        // Remove connection pointer
-        _server_index_lookup.erase(server_info.conn);
-
         // Destroying the connection will complete its outstanding operations
         delete server_info.conn;
 
         // Don't wait for any more completions on this connection
         _num_sent -= get_num_uncompleted(server_info);
 
+        // Remove connection pointer
+        // server_info.conn is used as ptr value, no real access to object
+        /* coverity[use_after_free] */
```

Comment:
better avoid coverity comments in example code

Reply:
ok, re-implemented